jquery.collection.js

```diff
@@ -655,6 +673,26 @@
             return elements;
         }
 
+        var getElementKey = function (settings, element) {
+            var name = element.find(':input[name^="' + settings.name_prefix + '["]').attr('name');
```

Comment:
On nested collections, `settings.name_prefix` contains `[` and `]` that will break the jquery expression.